cloud/ec2_client.go

```diff
@@ -146,29 +148,47 @@ func awsClientDefaultRetryOptions() utility.RetryOptions {
 	}
 }
 
+// configCache is a cache that maps a unique identifier for the AWS
+// configuration to the corresponding AWS configuration.
 var configCache map[string]*aws.Config = make(map[string]*aws.Config)
 
+func getConfigCacheID(role, region string) string {
+	return fmt.Sprintf("%s-%s", role, region)
+}
+
 // Create a new aws-sdk-client if one does not exist, otherwise no-op.
-func (c *awsClientImpl) Create(ctx context.Context, region string) error {
+func (c *awsClientImpl) Create(ctx context.Context, role, region string) error {
 	if region == "" {
 		return errors.New("region must not be empty")
 	}
 
-	if configCache[region] == nil {
-		config, err := config.LoadDefaultConfig(ctx,
-			config.WithRegion(region),
-		)
+	configID := getConfigCacheID(role, region)
+	if configCache[configID] == nil {
+		opts := []func(*config.LoadOptions) error{config.WithRegion(region)}
+		if role != "" {
+			// Assuming a role to make API calls requires an explicit region.
+			stsConfig, err := config.LoadDefaultConfig(ctx, config.WithRegion(evergreen.DefaultEC2Region))
```

Comment:
Should we use the passed in region (is guaranteed to be not ""?) or it doesn't really matter? I could see someone coming along and "fixing" it to use the provided region, maybe in the comment we could say `Assuming a role to make API calls requires specifying any region.` or something like that

Reply:
The region passed in is for the actual API call that's being made, not the credentials provider. This is based on the assumption that the runtime environments team will always set up the roles in us-east-1 (like why the STS manager does) rather than some other region. I guess we could explicitly add a region to the admin settings, but I don't imagine we're that close to a future where Evergreen has roles in a different region.

Reply:
Ah that makes sense, I wasn't aware of that assumption- I'll keep it in mind 🧠